pyfmt/__init__.py

```diff
@@ -25,8 +27,121 @@
+    """
+    Divider between sections of program.
+    """
+    if title:
+        print(
+            "\n"
+            + "{}".format(color_code)
+            + "  {}  ".format(title.upper()).center(
+                os.get_terminal_size().columns, "{}".format(character)
+            )
+            + "\033[0m"
+        )
+    else:
+        print(
+            "\n"
+            + "{}".format(color_code)
+            + "".center(os.get_terminal_size().columns, "{}".format(character))
+            + "\033[0m"
+        )
+
+
+def pyfmt(
+    path, skip="", check=False, line_length=100, extra_isort_args="", extra_black_args=""
+) -> int:
     """Run isort and black with the given params and print the results."""
+    pyfmt_title()  # Display title
+    timer_start = time.time()  # Measure how long everything takes
+
+    if skip:
+        # Map out current working directory
+        all_files, all_dirs = find_all_files_and_dirs()
+        # If specified files and directories exist, store the filenames
+        skips = skip.split(",")
+        filenames_to_skip = []
+        for item in skips:
+            if item in all_files:
+                if item.split(".")[-1] == "py":
+                    filenames_to_skip.extend([item])
+            elif os.path.abspath(item) in all_dirs:
+                # Saving all filenames in directory
+                files_in_dir = list()
+                for (dirpath, dirnames, filenames) in os.walk(item):
+                    for filename in filenames:
+                        if filename.split(".")[-1] == "py" or filename.split(".")[-1] == "pyi":
+                            files_in_dir.append(filename)
+                filenames_to_skip.extend(files_in_dir)
+            else:
+                print(
+                    'CRITICAL: One of the files or directories marked as skipped not found ("{}").'.format(
+                        item
+                    )
+                )
+                print("CRITICAL: Check spelling or existence of file or directory")
+                print("CRITICAL: Aborting pyfmt ...")
+                sys.exit()
+        # Display Files skipped
+        display_divider(title="SKIPPING FILES")
+        for filename_to_skip in filenames_to_skip:
+            print(f"SKIPPING: {filename_to_skip}")
+        print("\nNumber of files to be skipped: {}".format(len(filenames_to_skip)))
+        # Make a continuos string of arguments for
+        #   isort - must be separate --skip for each file
+        #   black - regex for exact filename (ie. file1|file2|etc.)
+        isort_filenames_to_skip = ""
+        black_filenames_to_skip = ""
+        for filename in filenames_to_skip:
+            isort_filenames_to_skip += "--skip=" + filename + " "
+            black_filenames_to_skip += filename + "|"
+        isort_filenames_to_skip = isort_filenames_to_skip[:-1]
+        black_filenames_to_skip = black_filenames_to_skip[:-1]
```

Comment:
why this?

Reply:
remove that last character, " " and "|" in each string ... probably a better way to do it

Reply:
ah yeah, you could do this instead:

```python
isort_filenames_to_skip = "--skip=" + " --skip=".join(filenames_to_skip)
black_filenames_to_skip = "--exclude=" + "|".join(filesnames_to_skip)
```

instead of the `for` loop above

Reply:
much better!